Add more description and example for identity

edwardloveall opened the pull request:
Hello! I took a try at adding content for Connascence of Identity. I saw there's already #10 but it hasn't been touched in a while and I had a different take on it inspired by [Jim Weirich's talk](https://vimeo.com/10837903) (at 37:51)

edwardloveall committed:
Add more description and example for identity

thomir commented:
hi @edwardloveall - thanks for the contribution here!

I haven't had the time I'd like to dedicate to this project, so apologies in advance if I'm a bit slow to reply 😅 

Regarding your submission, I'd love to dig in to the specific example used in your PR. My specific concern is that the example has two things potentially going on:

1. Connascence of value, if `User.find_one(type: "system")` returns a different user
2. Connascence of identity,  as you point out in the comments. 

I think one way to work around `1` is to simply replace `type: "system"` with something like: `username: "admin"`, which suggests to the reader that the return value will always represent the same system user, leaving only the connascence of identity as a problem. 

What do you think?

edwardloveall commented:
Thanks @thomir. No worries about reply time. I have some personal obligations this week so I also won't be able to respond quickly either 🙂 

I think I see the confusion. (Still wrapping my head around connascence of value.) In my head, there was only one user of `type: "system"` in the database, just as there would also be one user with `username: "admin"`. That's totally unclear though! 😅

Your suggestion to replace it with `username: "admin"` would help clarify. Another thing that might help is replacing the Rails-like user fetching code with placeholder so as not to distract:

```ruby
class SystemUser
  def self.get
    # find and instantiate the system user from the database
  end
end
```

or code that makes it clear that it's a new instance:

```ruby
class SystemUser
  def self.get
    data = fetch_db_row(table: "users", username: "system")
    User.new(data)
  end
end
```

Does this help disambiguate? Did I just commit a different (or the same 😱) connascence crime? 😄

thomir commented:
definitely - I think any of the following would work:

* `username: "admin"` works, since usernames are typically unique within a system.
* Replacing with a comment
* Something like `data = fetch_db_row(table: "users", username: "system")`

Personally, I like the last option the least, since it's more for the user to read, and might cause folks to focus on the wrong thing. I'm 👍 on either of the first two options though.

edwardloveall committed:
Switch system user implementation to a comment

So as not to distract from connascence of identity with database
implementation specifics.

edwardloveall commented:
Awesome, thanks for the feedback. Updated in [3fa92dc](https://github.com/connascence/websource/pull/13/commits/3fa92dc1520f2c9515f20e3c93a30ba565e73831)

edwardloveall commented:
Hi! I wanted to see if there were any improvements to this you thought I could make. Discussing what is "good" or "bad" is also pretty tricky so let me know if there's anything I can do to help. You mentioned you'd be slow to respond so if now's a bad time, let me know and I can come back later 🙂

thomir commented:
Apologies for the delay! This looks great - I'll merge it now.
